### PR TITLE
Add documentation for each icon

### PR DESCRIPTION
**Pull request by michaelspiss**
My proposal for a per-icon documentation. It includes the icon's name, style, url and its search terms if available. This is only a proposal, feel free to edit!

### CODE REVIEW

File: tool/generate_font.dart [hunk 1]
```diff
@@ -120,24 +123,50 @@ void main(List<String> arguments) {
...
 
-String generateIconDefinition(String iconName, String style, String unicode) {
+String generateIconDocumentation(
+    String iconName, String style, List searchTerms) {
+  searchTerms = searchTerms ?? [];
+  var searchTermsString = searchTerms.toString();
+  searchTermsString =
+      searchTermsString.substring(1, searchTermsString.length - 1);
+
+  iconName = iconName.replaceFirst("solid_", "");
```
Comment (brianegan) [inline tool/generate_font.dart:133]: Would we need to take into account any other styles for the pro icon sets such as `light_` or `duotone_`?

**Review by brianegan — APPROVED** [review]
Hey, thanks so much for tackling this one! Great work :)

File: tool/generate_font.dart (hunk 1 above)
Comment (michaelspiss) [reply tool/generate_font.dart:133]: They deviated from their naming scheme for some solid icons only. We could add preemptive checks for other styles, but I don't see the need for that

**Comment by michaelspiss** [comment]
Do you think the doc comments are comprehensive? Is there anything you would add/change?

File: tool/generate_font.dart (hunk 1 above)
Comment (brianegan) [reply tool/generate_font.dart:133]: Sounds good!

**Comment by brianegan** [comment]
> Do you think the doc comments are comprehensive?

Yeah, actually I thought adding the search terms was a really nice touch! The only other thing you could consider: Passing the `label` from the JSON through to `generateIconDefinition` and using that for the first line.

That would change

```

**Comment by michaelspiss** [comment]
I agree, this looks much cleaner now

**Pull request merged**